### PR TITLE
osutil: change escaping for create-user's sudoers

### PR DESCRIPTION
**Pull request by niemeyer**
This is applying the review on @mwhudson's #1651, and testing the feature.

### CODE REVIEW

File: osutil/user.go [hunk 1]
```diff
@@ -46,23 +55,38 @@ func AddExtraSudoUser(name string, sshKeys []string, gecos string) error {
...
 		"--extrausers",
 		"--disabled-password",
-		"--add_extra_groups", "sudo",
 		name)
 	if output, err := cmd.CombinedOutput(); err != nil {
 		return fmt.Errorf("adduser failed with %s: %s", err, output)
 	}
 
+	// Must escape "." as files containing it are ignored in sudoers.d.
+	sudoersFile := filepath.Join(sudoersDotD, "create-user-"+strings.Replace(name, ".", "%2E", -1))
```
Comment (niemeyer) [inline osutil/user.go]: This is the only change on top of @mwhudson's branch, besides adding tests for this which were lacking in the original.

File: osutil/user_test.go [hunk 2]
```diff
@@ -39,23 +39,46 @@ var _ = check.Suite(&createUserSuite{})
...
+
+	fs, _ := filepath.Glob(filepath.Join(mockSudoers, "*"))
+	c.Assert(fs, check.HasLen, 1)
+	c.Assert(filepath.Base(fs[0]), check.Equals, "create-user-karl%2Esagan")
+	bs, err := ioutil.ReadFile(fs[0])
+	c.Assert(err, check.IsNil)
+	c.Check(string(bs), check.Equals, `
+# Created by snap create-user
+
+# User rules for karl.sagan
```
Comment (mvo5) [inline osutil/user_test.go]: haha! brilliant :)

**Comment by mvo5** [comment]
Thanks! :+1:

**Comment by mwhudson** [comment]
Ah thanks, this is definitely a better idea that what I did!

**Pull request merged**